bundles/org.openhab.binding.nikobus/src/main/java/org/openhab/binding/nikobus/internal/handler/NikobusBaseThingHandler.java

```diff
@@ -38,8 +44,15 @@ protected NikobusBaseThingHandler(Thing thing) {
     public void initialize() {
         address = (String) getConfig().get(NikobusBindingConstants.CONFIG_ADDRESS);
         if (address == null) {
-            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.CONFIGURATION_ERROR, "Address must be set!");
-            return;
+            String addressPC = (String) getConfig().get(NikobusBindingConstants.CONFIG_ADDRESS_PC);
+            if (addressPC != null) {
+                address = addressPC.substring(2, 4) + addressPC.substring(0, 2);
```

Comment:
would suggest to make an abstract function that does the parsing of `CONFIG_ADDRESS_PC` 
i.e. 
`address = parseNikobusAddress(addressPC)`
where `parseNikobusAddress` is an abstract method and implement it in
1. `NikobusModuleHandler` for modules (actually move this code from `NikobusBaseThingHandler` to `NikobusModuleHandler`) since its same for all modules;
2. in `NikobusPushButtonHandler` for buttons (it's there already, just move it to `parseNikobusAddress`)

That way we would have the code where it belongs.

Reply:
done